lldb/tools/lldb-dap/lldb-dap.cpp

```diff
@@ -1062,6 +1063,16 @@ void request_breakpointLocations(const llvm::json::Object &request) {
         continue;
       if (line == end_line && column > end_column)
         continue;
+
+      // Make sure we are in the right file.
+      // We might have a match on line & column range and still
+      // be in the wrong file, e.g. for included files.
+      if (std::string_view(line_entry.GetFileSpec().GetFilename()) !=
```

Comment:
FWIW, these strings are coming from lldb's string pool, so you should be able to just compare the raw pointers for equality.